scenes/cslib_array.txt

```diff
@@ -0,0 +1,215 @@
+*bug cslib_array is a cslib module and must not be accessed directly
+
+*comment TODO: Decide array_attribute structure
+*comment *comment FILTER_BY_VALUE
+*comment *comment ------------------------------
+*comment *label filter_by_value
+*comment *params p_array_ref_src p_array_ref_tar p_value
+*comment *temp filter_by_routine false
+*comment *goto _filter_common
+
+*comment FILTER_BY_ROUTINE
+*comment ------------------------------
+*label filter_by_routine
+*params p_array_ref_src p_array_ref_tar p_scene p_label
+*temp filter_by_routine true
+*goto _filter_common
+
+*label _filter_common
+*temp src_count {p_array_ref_src&"_count"}
+*temp tar_max {p_array_ref_tar&"_max"}
+*set {p_array_ref_tar&"_count"} 0
+*temp src_index 1
+*temp tar_index 1
+*label _filter_by_routine_loop
+*if (src_index <= src_count)
+    *if (src_index > tar_max)
+        *bug cslib_array.filter_by_routine: Target array '${p_array_ref_tar}' of max size ${tar_max} is too small to hold all filtered values
+    *comment pass current element to filter function
+    *if (filter_by_routine)
```

Comment:
if we are not (yet) doing filter by value, I would simplify the code here - YAGNI   :-)

https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it